src/scheduler/mod.rs

```diff
@@ -106,8 +107,8 @@ impl<'a, R: 'a + Resource + Send + Sync> Scheduler<R> {
                         break;
                     }
                     Ok(Control::Finished(dir, task_ident)) => {
-                        scheduler.lock().map(|scheduler| {
-                            scheduler.scheduler_root.lock().map(|mut scheduler_root| {
+                        let _ = scheduler.lock().map(|scheduler| {
+                            let _ = scheduler.scheduler_root.lock().map(|mut scheduler_root| {
```

Comment:
Although not introduced here, the fact that this return value is not being used highlights that something is wrong with this code. The last `.map(|task_set|)` (below) seems to be superfluous. I *think* it can be safely removed and that it is left over from an earlier iteration, but we should make sure that's true (then delete if so).

Reply:
`lock()` never fails (At least it is assumed in our code that lock() never fails since we are doing `.lock().unwrap()` in many places) So consequently, this `.map()` will also never fail. I think this can be written in a cleaner way. Though let's do it in a separate PR.

Reply:
Agreed on not addressing in this PR.

My point is that the code on lines 116 and 119 won't have any effect. So it will never fail and never do anything either. I *think* it doesn't need to do anything with the current logic, in which case it can just be removed.